ops/jenkins/bfd-jenkins-sonarqube/Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env groovy
+
+// These variables are accessible throughout this file (except inside methods and classes).
+def verboseMaven = params.verbose_mvn_logging
+def appBuildResults
+
+pipeline {
+    agent {
+        kubernetes {
+            defaultContainer 'bfd-cbc-build'
+            yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  serviceAccount: bfd
+  containers:
+  - name: bfd-cbc-build
+    image: 'public.ecr.aws/c2o1d8s9/bfd-cbc-build:jdk17-mvn3-tfenv3-latest'
+    command:
+    - cat
+    tty: true
+    imagePullPolicy: Always
+    resources:
+      requests:
+        memory: '16384Mi'
+        cpu: '8000m'
+      limits:
+        memory: '16384Mi'
+        cpu: '8000m'
+"""
+        }
+    }
+
+    parameters {
+        booleanParam(name: 'verbose_mvn_logging',
+        description: 'When true, `mvn` will produce verbose logs.',
+        defaultValue: false)
+        }
+    environment {
+        SONAR_TOKEN = credentials('svc-sonar-bfd')
+        SONAR_PROJECT_KEY = 'bfd-parent'
+        SONAR_HOST_URL = 'https://sonarqube.cloud.cms.gov'
+    }
+    stages {
+
+            stage('Prepare') {
+                steps {
+                    script {
+                        // Load the child Jenkinsfiles.
+                        scriptForApps = load('apps/build.groovy')
+                        awsAuth.assumeRole()
+                    }
+
+               }
+
+            }
+        stage('Authenticate with AWS') {
+          steps {
+            script {
+              awsAuth.assumeRole()
+            }
+          }
+        }
+
+        stage('Checkout Stage') {
+            steps {
+               checkout scm
+            }
+        }
+        stage('Build Stage') {
+            // Build Step
+         steps {
+
+                script {
+                   // Perform Build steps
+                   awsAuth.assumeRole()
+                   appBuildResults = scriptForApps.build(verboseMaven)
+                }
+
+            }
+        }
+
+        stage ('SonarQube Analysis') {
+            steps {
+                script {
+                    withSonarQubeEnv('SonarQube Server') {
```

Comment:
I haven't made up my mind on this, but... I think it would be simple enough to go without `withSonarQubeEnv` and forego the [Jenkins extension for SonarQube](https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/jenkins-extension-sonarqube/).
1. I don't think this is fully leveraged just yet–if it were, I don't think the environment block above would be necessary, particularly the host and token definitions:
  ```groovy
//..snipped..
    environment {
        SONAR_TOKEN = credentials('svc-sonar-bfd')
        SONAR_PROJECT_KEY = 'bfd-parent'
        SONAR_HOST_URL = 'https://sonarqube.cloud.cms.gov'
//..snipped..
    }
  ```
  2. This isn't clearly dictated in the AC of the story, but... there's new and renewed interest in potentially migrating away from Jenkins. Even if that doesn't materialize, keeping the scripting in the Jenkinsfile resources as simple as possible, potentially even supporting copy-pastable content for operators to replicate the workflow locally _could_ be a win for maintainability, etc.